Add resource management alerts (#82)
appuio/component-openshift4-monitoring: docs/modules/ROOT/pages/runbooks/podcapacity.adoc

= Alert Group: syn-PodCapacity

An OpenShift cluster can handle a limited number of pods per node.
If you try to run more pods than this limit, OpenShift won't be able to schedule these pods.

> We provide two rules to notice if a cluster is hitting the limit of runnable pods.

Reviewer: ```suggestion
We provide two rules to notice if a cluster is hitting the limit of schedulable pods.
```

== Alert Rule: SYN_TooManyPods [[SYN_TooManyPods]]

=== icon:glasses[] Overview

This alert means that the cluster only has capacity for few additional pods.

> By default this threshold is the pod limit per node.

Reviewer: ```suggestion
By default, this threshold is the pod limit per node.
```

> This means if you receive this alert the cluster will likely won't be able to keep all workloads running in case of a worker node failure.

Reviewer: ```suggestion
If you receive this alert, it's likely that the cluster won't be able to keep all workloads running in case of a worker node failure.
```

> After verifying the alert you should add worker nodes as soon as possible.

Reviewer: ```suggestion
After verifying that the amount of running pods in the cluster is close to the cluster's pod capacity, you should add worker nodes as soon as possible.
```

=== icon:search[] Investigate

* Verify the alert
** Check the per worker node pod limit of your cluster
+
[source,shell]
----
kubectl get KubeletConfig -o yaml
----
** Check the number of pods on each worker node
+
[source,shell]
----
kubectl describe node -lnode-role.kubernetes.io/app | grep -E "(^Name:|^Non-terminated)"
----
** Verify that the reported capacity is correct.

> If not there might be a bug in the alert rule.

Reviewer: ```suggestion
If there's a discrepancy between the alert, and the actual number of running pods reported by Kubernetes, there might be a bug in the alert rule.
```
